### PR TITLE
Custom command to run benchmark from a PR

### LINKED ISSUE #59: Benchmark workflow does not work for PRs from forks
The Benchmark workflow is supposed to run the benchmarks and comment the resources in the PR. However, it needs permissions that can't be granted if the PR is from a fork.

The "straightforward but not simple solution" would be to change our workflow entirely and use branches in the main repo instead of forks.

@suvayu also posted the idea below:
 
> IMO this should be a bot, and a cron job.  Bot to manually trigger it in a PR, and cron that checks performance regression, say, once a week.

_Originally posted by @suvayu in https://github.com/TNO-Tulipa/TulipaBulb.jl/issues/50#issuecomment-1706483823_

@suvayu, do you know how to set that up, do you know if it works without permissions?

### PR DESCRIPTION
**Pull request by suvayu**
# Pull request details

## Describe the changes made in this pull request

This implements a `/run-benchmark` "command" that anyone can use in PRs to trigger the `Benchmark.yml` workflow.  After the workflow completes, it responds to the PR with a comment containing the results.  The results from `BenchmarkCI` doesn't seem like to be markdown formatted, so I wrapped it in a source block.  We could also try to parse it into nice markdown if we want.  The workflow also has an (incomplete) `workflow_dispatch` trigger, which could be used to manually run it for a specific branch/tag.  To support this use, the workflow also generates a "step summary" that is shown in the GH actions workflow run page.

You can see this in action in [this repo](https://github.com/suvayu/fictional-giggle/pull/5). In the example repo there's nothing to benchmark, so it just responds with the latest commit of the PR branch (this way you also know the correct version is being "benchmarked").

#### Complete:
- [x] trigger with custom command: `/run-benchmark`
- [x] benchmark report as PR comment & step summary

#### Incomplete:
- [ ] trigger on a branch/tag with `workflow_dispatch`

#### Possible improvements:
- [ ] limit to members
- [ ] link back from the comment to the workflow run page
- [ ] parse and format report as markdown

## List of related issues or pull requests

Closes #59

## Collaboration confirmation

As a contributor I confirm

- [x] I read and followed the instructions in README.dev.md
- [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
- [x] Tests are passing
- [x] Lint is passing

### CODE REVIEW

File: .github/workflows/Benchmark.yml [hunk 1]
```diff
@@ -23,7 +50,26 @@ jobs:
...
             - name: Run benchmarks
               run: julia -e 'using BenchmarkCI; BenchmarkCI.judge(baseline="origin/main")'
-            - name: Post results
-              run: julia -e 'using BenchmarkCI; BenchmarkCI.postjudge()'
-              env:
-                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            # - name: Post results
+            #   run: julia -e 'using BenchmarkCI; BenchmarkCI.postjudge()'
+            #   env:
+            #       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (abelsiqueira) [inline .github/workflows/Benchmark.yml:56]: Can be removed

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Nice. Thanks. I pointed out the left over comment for removal.
Also, can you add this info to the README.dev?

**Comment by suvayu** [comment]
@abelsiqueira should I address the "incomplete/improvements" items in this PR, or leave it for later?

**Comment by suvayu** [comment]
> If we run with workflow_dispatch, can we see the results somewhere? (E.g., the terminal?)

Currently, it's incomplete.  But every benchmark run also shows the output in workflow run page.  For example see this one from the [example repo](https://github.com/suvayu/fictional-giggle/actions/runs/6214765185).  Basically whatever you would see in the comments, is replicated at the end as a step summary

**Comment by abelsiqueira** [comment]
Then I think the solution here is good enough and we can work on improvements after using it a bit

**Review by abelsiqueira — APPROVED** [review]
Thanks :beach_umbrella:

**Comment by suvayu** [comment]
Damn! Looks like only owners can merge, so you'll have to merge it :smirk: @abelsiqueira

**Comment by abelsiqueira** [comment]
Done. Thank you.

**Pull request merged**